src/components/MailingList/MailingList.jsx

```diff
@@ -30,51 +34,106 @@ const EmailSubmitButton = styled.input`
+statusToTitle[LOADING_STATUS] = "Sending...";
+statusToTitle[SUCCESS_STATUS] = "Sent!";
+statusToTitle[FAILURE_STATUS] = "Failed to send";
 
-    // handle error
-    if (response.status !== 200) {
-      setResult(`An error occured: ${response.status} ${response.statusText}`);
-      return;
-    }
+export default function MailingList() {
+    const [address, setAddress] = useState("");
+    const [status, setStatus] = useState(READY_STATUS);
+    const [disabled, setDisabled] = useState(false);
 
-    // handle successful response
-    setAddress("");
-    setResult("Successfully added!");
-  }
+    async function addEmailToList(e) {
+        e.preventDefault();
+        setStatus(LOADING_STATUS);
+        setDisabled(true);
+        //make email post request
+        const response = await fetch("https://api.michhackers.com/email/add", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify({
+                email: address,
+            }),
+        });
+
+        // handle error
+        if (response.status !== 200) {
+            setStatus(FAILURE_STATUS);
+            setDisabled(false);
+            console.error(
+                `An error occured: ${response.status} ${response.statusText}`
+            );
+            return;
+        }
+        setStatus(SUCCESS_STATUS);
+        setAddress("");
+        setDisabled(false);
+    }
 
-  return (
-    <EmailForm onSubmit={addEmailToList}>
-      {result && <StaticP>{result}</StaticP>}
-      <EmailInputBox
-        type="email"
-        value={address}
-        onChange={e => setAddress(e.target.value)}
-        placeholder="michiganhackers@umich.edu"
-        required
-      />
-      <EmailSubmitButton type="submit" value="Join" />
-    </EmailForm>
-  );
+    return (
+        <EmailForm onSubmit={addEmailToList}>
+            <EmailInputBox
+                type="email"
+                value={address}
+                onChange={(e) => {
+                    setAddress(e.target.value);
+                    setStatus(READY_STATUS);
+                    setDisabled(false);
+                }}
+                placeholder="michiganhackers@umich.edu"
+                required
+            />
+            <EmailSubmitButton
+                type="submit"
+                disabled={disabled}
+                title={statusToTitle[status]}
+            >
+                <LoadingAnimation active={status === LOADING_STATUS}>
+                    {status === LOADING_STATUS ? (
+                        <FontAwesomeIcon icon={["fas", "spinner"]} />
+                    ) : status === READY_STATUS ? (
+                        <FontAwesomeIcon icon={["fas", "paper-plane"]} />
+                    ) : status === SUCCESS_STATUS ? (
+                        <FontAwesomeIcon icon={["fas", "check"]} />
+                    ) : (
+                        <FontAwesomeIcon icon={["fas", "times"]} />
```

Comment:
I think a restart icon would be more effective in telling the user to try resubmitting.